packages/azure-http-specs/specs/azure/resource-manager/non-resource/mockapi.ts

```diff
@@ -0,0 +1,30 @@
+import { json, passOnSuccess, ScenarioMockApi } from "@typespec/spec-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+const SUBSCRIPTION_ID_EXPECTED = "00000000-0000-0000-0000-000000000000";
+const LOCATION_EXPECTED = "eastus";
+
+const nonResource = {
+  id: "id",
+  name: "hello",
+  type: "nonResource"
+};
+
+Scenarios.Azure_ResourceManager_NonResource_NonResourceOperations_get = passOnSuccess({
+  uri: "/subscriptions/:subscriptionId/providers/Microsoft.NonResource/locations/:location/otherParameters/:parameter",
```

Comment:
Why does the URL look exactly like a resource? Does this reflect the real use case?

Reply:
This test is only for validating that SDK can generate without reporting error, and we don't have requirements on how SDK looks, right?

Reply:
I'm not sure I follow you. What's the problem it looks like a resource url? As long as they define it as a non resource operation, we should take it as a non resource operation, no matter how it looks like. 

Same as the model, no matter how it looks like a real resource. As long as user defines it as non resource model, we should take it as a non resource model.

As yes, it is a real case.

Reply:
How SDK looks like is each language's own decision. I can only speak to how C# looks like. You decide how Java looks like, etc...

Reply:
To give you more insights, TypeSpec's advantages over swagger is that it gives clear information about what is a resource, what is not a resource, instead of the guessing logic in swagger. We definitely should honor such information to unveil such advantages. So I add these tests.

Reply:
Thanks. Currently Java hasn't taken advantage of the TypeSpec information.